uktrade/data-hub-frontend
# Fix/manage randomness in tests

## Description of change

This PR prevents direct import of `faker` and `json-schema-faker` and only allows usage of their seeded, _pseudo-random_ instances from `test/sandbox/utils/random.js` to ensure tests are deterministic.

The only reason why the `utils/random.js` module is located undeer `test/sandbox` is that it's the only place from which it can be imported both from within the _sandbox_ sub-project and the rest of the codebase.

All tests seem to be consistently passing when I rerun the whole pipeline 10 times. In the 3 failed runs, only component tests failed because of a [bug in Cypress component test mode](https://github.com/cypress-io/cypress/issues/28644), which is not in any way related to randomness.
<img width="1300" alt="Screenshot 2024-12-04 at 12 14 02" src="https://github.com/user-attachments/assets/495e022a-3fdf-4d6a-906a-155d6a7cebbf">

## Test instructions

1. Run a test which makes use of `faker` or `json-schema-faker` (one of the many fakers that we have)
2. No matter how many times you run the test, the "random" value generated by these utilities should be identical between test runs
3. Importing `faker` or `json-schema-faker` directly should be a linter error
